### PR TITLE
Metadata match

### PR DESCRIPTION
**Pull request by florian-huber**
Starting to sketch a `MetadataMatch` similarity class.

### CODE REVIEW

File: matchms/similarity/MetadataMatch.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+        Metadata match between 2 and 3 is 1.00
+        Metadata match between 2 and 4 is 0.00
+
+    """
+    # Set key characteristics as class attributes
+    is_commutative = True
+    score_datatype = bool
+
+    def __init__(self, field: str,
+                 matching_type: str = "equal_match",
```
Comment (hechth) [inline matchms/similarity/MetadataMatch.py:68]: Wouldn't the easiest thing here be to actually pass a comparison function which has to evaluate to true and takes two values which are compared?

Comment (florian-huber) [reply matchms/similarity/MetadataMatch.py:68]: That would indeed work as well.
Only issue I see is that this cannot generally be optimized and will only work for pair-wise comparisons. For the two method given here I could at least do some numby optimization which I just pushed.

We could also do a combination here. So, offer the two default types I sketched and also allow passing a custom function. Would that work?

Comment (hechth) [reply matchms/similarity/MetadataMatch.py:68]: Could we restrict the functions to things which can be parallelized or so? I thought about passing like `==` operator, a `<=` etc. ... but maybe having both options of taking a some keywords which have numba implementations and allowing to pass a lambda with simple parallel for would be good. Btw., has matchms any parallelism, like parallel for loops at the moment?

Comment (florian-huber) [reply matchms/similarity/MetadataMatch.py:68]: So far parallel computation is depending on the respective scores. By default, no parallel execution will happen. I tried several things, but so far I haven't found a good way to implement parallel execution in Python that will work for all scores and/or all systems. The options which tend to work on all scores/systems can create huge bottlenecks and sometimes make things much slower than without parallel execution (e.g. using Python multiprocessing, or even dask).

Comment (hechth) [reply matchms/similarity/MetadataMatch.py:68]: Okay - I have little experience with parallelism in python, so likely it is better to keep that discussion somewhere else.

Comment (florian-huber) [reply matchms/similarity/MetadataMatch.py:68]: Yes, we have still the issue #165, but could also update that at some point.

**Comment by hechth** [comment]
@florian-huber This looks pretty good so far and can also be used somewhat independently from the other things. What do you think would still be required to make this a full-fledged PR? Because this would likely already implement #251 .

Let me know if I can contribute to this in any way 👍

**Comment by florian-huber** [comment]
@hechth, good  you mentioned #251. I agree that this PR should fix this issues. You also mentione a more flexible comparison function that will take custom functions as input. I would leave this out of this PR and either add it later, or (preferentially) create an entirely new score for this `CustomSimilarity` or similar which will take any user function of the form `custom_function(spectrum1, spectrum2) --> score`.

Let me know if this PR can be merged or if you want any further changes for now.

**Review by hechth — APPROVED** [review]
I think this is good for now and can be merged - we can always build upon it later. I think it would be best to do things step by step 👍

**Comment by florian-huber** [comment]
Thanks a lot for the review and suggestions Helge!

**Pull request merged**